modules/swagger-codegen/src/main/resources/rust/model.mustache

```diff
@@ -28,19 +29,28 @@ impl {{classname}} {
   }
 
   {{#vars}}
+  #[allow(non_snake_case)]
```

Comment:
@wing328 Can we avoid this? I know that other Codegen languages transform names according to the guidelines of the target programming language.

Reply:
I agree, I'm just new to swagger-codegen and wasn't sure how to get names exactly right.

In this case, I need two versions of the name, one with a leading underscore and one without, because the warning is due to having two underscores in a row. Do you know how to create two mustache variables, one with a leading underscore for reserved names and one without?

Reply:
@frol Is there any lib in Rust that can handle JSON (de)serialization?

@ahmedcharles please share the spec of the model.

Reply:
@wing328 [Serde](https://github.com/serde-rs/serde) is used here for JSON (de)serialization.

@ahmedcharles You can just use `_{{ var_name }}` in the template if you need a leading underscore.

Reply:
@frol I had a look at https://serde.rs/examples.html but couldn't an example how to provide a 1-1 mapping for attribute name (similar to what've done for Java, C#, etc). Do you know how we can map attribute name with Serde?

Reply:
@frol The issue with the warning is that `type` is a reserved word, so the `{{name}}` is `_type`, which when combined with `with_{{name}}` becomes `with__type`, which is what creates the warning. What I need is `{{name}}` = `_type` and `{{rawName}}` = `type`, so then I can do `fn {{name}}(...)` and `fn with_{{rawName}}(...)` and never have it generate names with two underscores in a row.

@wing328 Isn't that what `#[serde(rename = "jsonName")] rust_name: String;` does?

Reply:
@ahmedcharles I think so. I missed that.